base/irrationals.jl

```diff
@@ -13,7 +13,22 @@ promote_rule{s,T<:Number}(::Type{Irrational{s}}, ::Type{T}) = promote_type(Float
 convert(::Type{AbstractFloat}, x::Irrational) = Float64(x)
 convert(::Type{Float16}, x::Irrational) = Float16(Float32(x))
 convert{T<:Real}(::Type{Complex{T}}, x::Irrational) = convert(Complex{T}, convert(T,x))
-convert{T<:Integer}(::Type{Rational{T}}, x::Irrational) = convert(Rational{T}, Float64(x))
+
+@generated function convert{T<:Integer}(::Type{Rational{T}}, x::Irrational)
```

Comment:
Maybe I am mistaken but this doesn't seem to return an expression like a `@generated` function should. It also doesn't really seem to need to be a `@generated` function at all.